app/assets/javascripts/discourse/lib/to-markdown.js.es6

```diff
@@ -0,0 +1,211 @@
+import parseHTML from 'discourse/helpers/parse-html';
+
+String.prototype.trimLeft = function() {
+    return this.replace(/^\s+/,"");
+}
+
+const Tag = Ember.Object.extend({
```

Comment:
I don't think we need to use the Ember object model for this, since you are never observing or using computed properties. Using a regular Javascript class (or ES6 class) will be significantly faster.

I see you use `extend` below to set up the properties, but that should be possible via a constructor instead.